src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,281 @@
+        sys.exit(1)
+
+    script_args = parser.parse_args()
+
+    return script_args
+
+filename = '''binary-ops-{NUMBER}.js'''
+MAX = (1<<53) - 1
+MIN = -1 * MAX
+BITMAX = 9999999
+BITMIN = -9999999
+
+class GenSequence:
+    def __init__(self, options, bit):
+        num_operands = options.operand_count[0]
+        self.num_operands = num_operands
+        self.operators_list = []
+        for i in range(len(options.op)):
+            if (bit == False):
+                if (options.op[i] == "+"):
+                    self.operators_list.append(self.add)
+                if (options.op[i] == "-"):
+                    self.operators_list.append(self.sub)
+                if (options.op[i] == "*"):
+                    self.operators_list.append(self.mul)
+                if (options.op[i] == "/"):
+                    self.operators_list.append(self.div)
+                if (options.op[i] == "%"):
+                    self.operators_list.append(self.mod)
+            else:
+                if (options.op[i] == "^"):
+                    self.operators_list.append(self.bitxor)
+        self.operands = [0] * num_operands
+        self.operators = [0] * (num_operands -1)
+
+        if (bit == False):
+            for i in range(num_operands -1):
+                self.operands[i] = random.randint(MIN, MAX)
+                self.operators[i] = random.randint(0, len(self.operators_list) -1)
+            self.operands[num_operands -1] = random.randint(MIN, MAX)
+
+        else:
+            for i in range(num_operands -1):
+                self.operands[i] = random.randint(BITMIN, BITMAX)
+                self.operators[i] = random.randint(0, len(self.operators_list) -1)
+            self.operands[num_operands -1] = random.randint(BITMIN, BITMAX)
+
+    def add(self, op1, op2):
+        sum = op1 + op2
+        if sum < MIN:
+            op2 = MIN - random.randint(MIN, op1)
+            sum = op1 + op2
+        elif sum > MAX:
+            op2 = MAX - random.randint(op1, MAX)
+            sum = op1 + op2
+        return {'oper': '+', 'op2': op2, 'sum': sum}
+
+
+    def sub(self, op1, op2):
+        sum = op1 - op2
+        if sum < MIN:
+            op2 = - (MIN - random.randint(MIN, op1))
+            sum = op1 - op2
+        elif sum > MAX:
+            op2 = - (MAX - random.randint(op1, MAX))
+            sum = op1 - op2
+        return {'oper': '-', 'op2': op2, 'sum': sum}
+
+    def mul(self, op1, op2):
+        sum = op1 * op2
+        if sum < MIN:
+            op2 = MIN / op1
+        elif sum > MAX:
+            op2 = MAX / op1
+        if op2 > 0:
+            op2 = math.floor(op2)
+        else:
+            op2 = math.ceil(op2)
+        sum = op1 * op2
+        return {'oper': '*', 'op2': op2, 'sum': sum}
+
+    def div(self, op1, op2):
+        if op2 == 0:
+            op2 = 2
+        sum = op1 / op2
+        if sum > 0:
+            sum = int(math.floor(sum))
+        else :
+            sum = int(math.ceil(sum))
```

Comment:
Why do we need to round this values?

Reply:
Because if it might be a float, and `sum` might jump out of the interval if `op2` is not rounded.

Reply:
I think `int(sum)` is more than enough, hence it "cuts" the fraction part, so the result cannot be bigger then the end of the interval.

Reply:
This request does not seem resolved for me.

Reply:
Did you check if the `int(sum)` is enough? What does the standard says?

Reply:
_class int(x, base=10)_

Return an integer object constructed from a number or string x, or return 0 if no arguments are given. If x is a number, it can be a plain integer, a long integer, or a floating point number. **If x is floating point, the conversion truncates towards zero.** If the argument is outside the integer range, the function returns a long object instead.

This snippet is part of the python API [docs](https://docs.python.org/2/library/functions.html#int).

So it think `int(sum)` is still suitable for this case.